src/render/tree/mod.rs

```diff
@@ -199,74 +198,74 @@ impl Display for Tree {
         let root = self.root();
         let level = self.level();
         let theme = ui::get_theme();
-        let prune = self.context().prune;
-        let mut output = String::from("");
-
-        #[inline]
-        fn extend_output(output: &mut String, node: &Node, prefix: &str) {
-            output.push_str(&format!("{prefix}{node}\n"));
+        let size_left = self.context().size_left;
+
+        fn extend_output(
+            f: &mut Formatter,
+            node: &Node,
+            prefix: &str,
+            size_left: bool,
+        ) -> fmt::Result {
+            if size_left {
+                // write!(f, "{prefix}")?;
+                node.display_size_left(f, prefix)?;
+                writeln!(f, "")
+            } else {
+                // write!(f, "{prefix}")?;
+                node.display_size_right(f, prefix)?;
+                writeln!(f, "")
+            }
         }
 
-        #[inline]
         fn traverse(
```

Comment:
Same comment as before about inlining

Reply:
This function is directly recursive -- it cannot be inlined without some transformation that removes the recursion.

Assuming that such a transformation exists (and the compiler finds it), it's only called in one place (aside from the recursive case). The choice of whether or not to inline would not affect performance -- a single function call is not measureable.